sympy/core/mul.py

```diff
@@ -1130,6 +1130,7 @@ def _eval_is_positive(self):
 
         sign = 1
         saw_NON = False
+        not_real = 0
```

Comment:
not_real is misleading because you are not testing whether the argument is real or not. I would recommend the following:

```diff
diff --git a/sympy/core/mul.py b/sympy/core/mul.py
index a7cf275..9e00c9d 100644
--- a/sympy/core/mul.py
+++ b/sympy/core/mul.py
@@ -1129,8 +1129,7 @@ def _eval_is_positive(self):
         """

         sign = 1
-        saw_NON = False
-        not_real = 0
+        saw_NON = saw_NOT = False
         for t in self.args:
             if t.is_positive:
                 continue
@@ -1145,16 +1144,16 @@ def _eval_is_positive(self):
                 saw_NON = True
             elif t.is_positive is False:
                 sign = -sign
-                saw_NON = True
-                not_real += 1
+                if saw_NOT: return
+                saw_NOT = True
             elif t.is_negative is False:
-                saw_NON = True
-                not_real += 1
+                if saw_NOT: return
+                saw_NOT = True
             else:
                 return
-        if sign == 1 and saw_NON is False and not_real == 0:
+        if sign == 1 and saw_NON is False and saw_NOT is False:
             return True
-        if sign < 0 and not_real <= 1:
+        if sign < 0:
             return False

     def _eval_is_negative(self):
```